### ISSUE TITLE
[Windows] PrintJob::addPage() cause crash with Alt+Enter fullscreen and BITMAP printMethod

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#246]
## Problem Description
`PrintJob::addPage()` method call cause application crash when you try to print anything with `BITMAP` `printMethod` at Alt+Enter fullscreen state. Alt+Enter fullscreen will be "enabled" only with `<renderMode>direct</renderMode>` or `<renderMode>gpu</renderMode>` in application manifest.
Crashes occurs if you use `new PrintJobOptions(true)` with `true` value for `printAsBitmap` parameter/property.
Also crash occurs if you use `PrintJobOptions::printMethod = PrintMethod.BITMAP`.

Tested with different AIR SDKs versions even with latests AIR 32.0.0.89, AIR 33.0.2.338 and AIR 33.1.0.50 with multiple Windows devices with different OS versions with 32-bit and 64-bit AIR applications with captive runtime with different printers.
Same problem in all cases.
Printer doesn't matter - you can use for test even virtual PDF printer or something like that.
Content for printing doesn't matter.

It happens only if AIR application in Alt+Enter fullscreen state. With "classic" AIR fullscreen state via `Stage::displayState` there is no such crashes.
Some our users and third-party remote desktop services uses such fullscreen state (Alt+Enter) for some reasons.
We have no way to prevent or override Alt+Enter state. There is feature request for that: https://github.com/Gamua/Adobe-Runtime-Support/issues/24
Also I still didn't find a way to prevent Alt+Enter even with native extension.

It works fine with `PrintMethod.VECTOR` or `PrintMethod.AUTO` print methods (see **Workaround** below).

## Steps to Reproduce
1) Launch code below with Windows device. AIR application manifest must contain `<renderMode>direct</renderMode>` or `<renderMode>gpu</renderMode>`.
2) Press Alt+Enter to go to fullscreen state.
3) Click anywhere on stage. Application try to print simple page (`Sprite`).

Application example with sources attached.
[printjob_addpage_fullscreen_crash.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/4271052/printjob_addpage_fullscreen_crash.zip)

```
package {
	import flash.display.Sprite;
	import flash.events.Event;
	import flash.events.MouseEvent;
	import flash.printing.PrintJob;
	import flash.printing.PrintJobOptions;
	import flash.printing.PrintJobOrientation;
	import flash.printing.PrintMethod;
	
	public class PrintJobAddPageFullscreenCrash extends Sprite {
		
		public function PrintJobAddPageFullscreenCrash() {
			addEventListener(Event.ADDED_TO_STAGE, addedToStage);
		}
		
		private function addedToStage(e:Event):void {
			removeEventListener(Event.ADDED_TO_STAGE, addedToStage);
			
			stage.addEventListener(MouseEvent.CLICK, print);
		}
		
		private function print(e:MouseEvent):void {
			if (PrintJob.isSupported){
				var pj:PrintJob = new PrintJob();
				pj.jobName = "Print job";
				
				//Print anything - it doesn't matter for this issue
				var s:Sprite = new Sprite();
				s.graphics.beginFill(0xCCCCCC);
				s.graphics.drawRect(0, 0, 100, 100);
				s.graphics.endFill();
				
				if (pj.start2(null, false)){//showPrintDialog=true cause the same crash
					var pjo:PrintJobOptions = new PrintJobOptions(true);//This line cause crash
					//var pjo:PrintJobOptions = new PrintJobOptions(false);//This line works fine
					
					//pjo.printMethod = PrintMethod.BITMAP;//This line also cause crash
					//pjo.printMethod = PrintMethod.VECTOR;//This line prevent crash
					//pjo.printMethod = PrintMethod.AUTO;//This line prevent crash
					
					try {
						pj.addPage(s, null, pjo);//Application crashes at this line
					} catch(e:Error) {
						trace("printing error");
					} finally {
						pj.terminate();
					}
				}
			} else {
				trace("PrintJob not supported");
			}
		}
	}
}
```

**Actual Result:**
Application crashes with `Adobe AIR.dll` Application Error:
```
Problem signature:
  Problem Event Name:	BEX
  Application Name:	adl.exe
  Application Version:	33.1.0.50
  Application Timestamp:	5e4e2cf2
  Fault Module Name:	Adobe AIR.dll
  Fault Module Version:	33.1.0.50
  Fault Module Timestamp:	5e4e3637
  Exception Offset:	000d3df7
  Exception Code:	c0000409
  Exception Data:	00000007
  OS Version:	6.3.9600.2.0.0.256.48
  Locale ID:	1049
  Additional Information 1:	91ff
  Additional Information 2:	91ff7319b73b59061cbe0ef148a0c790
  Additional Information 3:	2c04
  Additional Information 4:	2c04b61585e7ba2f9bed57378880cf33
```

**Expected Result:**
Application correctly print specified `Sprite` pages without crashes.

## Known Workarounds
Use
`new PrintJobOptions(false)`
or 
`PrintJobOptions::printMethod = PrintMethod.VECTOR`
or
`PrintJobOptions::printMethod = PrintMethod.AUTO`
but in some cases (depend on printer/content) print quality could be different or lower with such settings.

**Comment by itlancer** [comment]
One important notice.
Alt+Enter fullscreen will be "enabled" only with `<renderMode>direct</renderMode>` or `<renderMode>gpu</renderMode>` in application manifest.

**Comment by itlancer** [comment]
Crash still exists with latest AIR 33.1.1.533:
```
Application crashes with: printjob_addpage_fullscreen_crash.exe, version: 0.0.0.0, time stamp: 0x60cc9d53
Faulting module name: Adobe AIR.dll, version: 33.1.1.533, time stamp: 0x60d48fda
Exception code: 0xc0000409
Fault offset: 0x00081537
Faulting process ID: 0x3250
Faulting application start time: 0x01d77745d15eff26
Faulting application path: C:\printjob_addpage_fullscreen_crash\printjob_addpage_fullscreen_crash.exe
Faulting module path: C:\printjob_addpage_fullscreen_crash\Adobe AIR\Versions\1.0\Adobe AIR.dll
Report ID: c64b9571-17c1-453c-a737-c979b2d02319
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by itlancer** [comment]
Fixed with latest AIR 33.1.1.575.
Thanks!
